### PR TITLE
MPL in GroundwaterFlow process

### PR DESCRIPTION
**Pull request by TomFischer**
Use MPL data structures in GroundwaterFlow process.

1. [x] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.3.0)
2. [x] Tests covering your feature were added? - Existing test are adapted.
3. [x] Any new feature or behavior change was documented? - There isn't any new feature.
4. [x] [Script](https://github.com/ufz/ogs/files/4257122/convertGroundWaterFlowToMPL.py.txt) for automatic prj file conversion.

### CODE REVIEW

File: ProcessLib/GroundwaterFlow/GroundwaterFlowFEM.h [hunk 1]
```diff
@@ -121,13 +97,28 @@ class LocalAssemblerData : public GroundwaterFlowLocalAssemblerInterface
         ParameterLib::SpatialPosition pos;
         pos.setElementID(_element.getID());
 
+        auto const& medium =
+            _process_data.media_map->getMedium(_element.getID());
```
Comment (endJunction) [inline ProcessLib/GroundwaterFlow/GroundwaterFlowFEM.h]:
```suggestion
            *_process_data.media_map->getMedium(_element.getID());
```

File: ProcessLib/GroundwaterFlow/GroundwaterFlowProcess.cpp [hunk 2]
```diff
@@ -18,6 +18,27 @@ namespace ProcessLib
...
 {
+
+void checkMPLProperties(MeshLib::Mesh const& mesh,
+                        GroundwaterFlowProcessData const& process_data)
+{
+    DBUG("Check the media properties of HT process ...");
+
+    std::array const requiredPropertyMedium = {
+        MaterialPropertyLib::reference_temperature,
+        MaterialPropertyLib::PropertyType::diffusion};
```
Comment (endJunction) [inline ProcessLib/GroundwaterFlow/GroundwaterFlowProcess.cpp]:
```suggestion
        MaterialPropertyLib::PropertyType::reference_temperature,
        MaterialPropertyLib::PropertyType::diffusion};
```

**Comment by endJunction** [comment]
Looks good.
Didn't see the `diffusion` prj file tag docu...
After the tests are fixed, :fast_forward:

**Comment by TomFischer** [comment]
> Didn't see the `diffusion` prj file tag docu...
In the prj file the property diffusion is specified by the name-tag. How should the content of the tag be documented? I didn't find documentation for all the other values of the enum PropertyType.

**Comment by endJunction** [comment]
Never mind the tag docu. Was confused and thought it would be a new property.

**Comment by endJunction** [comment]
TestQtPrjInterface.QtXmlPrjReader is still failing: https://jenkins.opengeosys.org/blue/organizations/jenkins/ufz%2Fogs/detail/PR-2820/8/pipeline/62

**Comment by TomFischer** [comment]
> TestQtPrjInterface.QtXmlPrjReader is still failing: https://jenkins.opengeosys.org/blue/organizations/jenkins/ufz%2Fogs/detail/PR-2820/8/pipeline/62

I' m already in discussion with @rinkk rinkk to fix the issue.

**Pull request merged**